### ISSUE TITLE
PCRE (8.44) consumed in Erlang is vulnerable CVE-2017-11164

### ISSUE THREAD

**Issue by graorane** [erlang/otp#7464]
Erlang is consuming PCRE version 8.44.
https://github.com/erlang/otp/blob/c5b1f6e949f45151ae89f204102c06b4140922cb/erts/emulator/pcre/local_config.h#L86

This PCRE version as well as it's subsequnet version 8.45 is vulnerable. CVE-2017-11164, reference link as below.
https://nvd.nist.gov/vuln/detail/cve-2017-11164

PCRE is now end of life and it is replaced with PCRE2.
https://www.pcre.org/
Copying relevant text from above link

-------------
Versions
There are two major versions of the PCRE library. The current version, PCRE2, released in 2015, is now at version 10.39.

The older, but still widely deployed PCRE library, originally released in 1997, is at version 8.45. This version of PCRE is now at end of life, and is no longer being actively maintained. Version 8.45 is expected to be the final release of the older PCRE library, and new projects should use PCRE2 instead.
-------------

It seems that CVE-2017-11164 is not applicable to PCRE2.

So there are two queries
1.  Is there plan to update PCRE2 in Erlang?**
2.  Will it be possible to build existing versions of Erlang (26.X, 25.x etc.) by replacing PCRE with PCRE2?

**Comment by jhogberg** [comment]
Thanks for your report. It does not look like we're affected: `pcre` needs to be compiled for on-stack recursion for this problem to occur, and we compile for on-heap recursion instead.

> 1. Is there plan to update PCRE2 in Erlang?**

No, in OTP 27 we are going to replace it with [`re2`](https://github.com/google/re2) instead.

> 2. Will it be possible to build existing versions of Erlang (26.X, 25.x etc.) by replacing PCRE with PCRE2?

No.

**Comment by graorane** [comment]
Thanks a lot for the response.

**Comment by graorane** [comment]
Hi
I can still see PCRE 8.44 in OTP 27. 
https://github.com/erlang/otp/tree/OTP-27.0/erts/emulator/pcre
https://github.com/erlang/otp/blob/OTP-27.0/erts/emulator/pcre/pcre.h

So is it still getting consumed in Erlang?

**Comment by ssm6498** [comment]
Any update on this? is PCRE (8.44) still getting consumed in Erlang? If not please share release build version to refer.

**Comment by garazdawi** [comment]
You can get the current pcre version used by calling:

```erlang
Erlang/OTP 27 [erts-15.0] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:ns]

Eshell V15.0 (press Ctrl+G to abort, type help(). for help)
1> re:version().
<<"8.44 2020-02-12">>
```

so yes, pcre version 8.44 is still used, but the fixes in 8.45 do still not apply to our usage of it.

We still plan to change RE engine to something that is maintained, but it did not make it to Erlang/OTP 27.
